model/resource/service_experimental.yaml

```diff
@@ -22,16 +22,116 @@ groups:
+          generate a `service.instance.id`. Future versions of the OpenTelemetry Operator might also have the ability to
+          generate the `service.instance.id` and set container-specific labels, given it is in a position to know all that
+          metadata. When SDKs don't have sufficient data to generate a stable UUID, a random one is generated.
+
+          In more concrete terms: users running their services on platforms such as Kubernetes are encouraged to explicitly
+          set the `service.instance.id` using their existing automation, or set a value that can be used for a
+          consistent value, such as `container.id`. Similarly, users of application servers such as `unicorn`
+          are encouraged to set the `service.instance.id` on a per-worker basis. SDKs MUST use the following algorithm when
+          generating `service.instance.id`:
+
+          - If the user has provided a `service.instance.id`, via environment
+            variable, configuration or custom resource detection, this MUST take priority over generated IDs.
+          - When any of the below combinations of resource attribute are provided, they MUST be used as the input
+            for generating a UUID v5 following the prefix mentioned above. The values within each combination MUST be separated with dots:
+            * `container.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${container.id}`,
+            possibly without the namespace.
+            * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`, resulting in the input
+            `${telemetry.sdk.name}.${telemetry.sdk.language}.${k8s.namespace.name}.${service.name}.${k8s.pod.name}.${k8s.container.name}`. In this case,
+            the namespace MUST be used.
+            * `host.id`, resulting in the input `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${host.id}`,
+            possibly without the namespace.
+          - When the SDK is running in an environment where a `/etc/machine-id`
+            (see [MACHINE-ID(5)](https://www.freedesktop.org/software/systemd/man/machine-id.html))
+            is available, the machine-id SHOULD be used in the input for generating a UUID v5 (possibly without the namespace):
+            `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}.${machine.id}`
+          - When the SDK is running on a Windows environment and there's a reasonable way to read
+            registry keys for the SDK, the registry key
+            `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography\MachineGuid` can be used in a
+            similar way to Linux' machine-id above.
+          - When no other source is available the SDK MUST generate a value using UUID v1 or v4.
+            This would also typically be the case for service instances running on Kubernetes,
```

Comment:
"This would typically be the case for service instances running Kubernetes".

I don't think this is true.  MOST non-k8s environments may also struggle with this requirement.

Reply:
I removed the "also", but not sure about the second part: this extra information about Kubernetes is in result of previous comments here, as we cannot use a single service.instance.id for all containers within the same pod. Most non-K8s environments may also struggle, but the statement about Kubernetes doesn't contradict it.

Reply:
Marking as resolved as it's using the suggested wording. Thank you for the suggestion!